pydantic/tools.py

```diff
@@ -0,0 +1,20 @@
+from functools import lru_cache
+from typing import Any, Type, TypeVar
+
+__all__ = ('parse_as_type',)
+
+
+@lru_cache(maxsize=None)
```

Comment:
`maxsize=None` a bit dangerous? Maybe better to just use a large maxsize?

Reply:
I want to use this function when generating API clients, so I would expect it to be used for every possible response model from a given API. Given the usefulness of inheritance when defining slightly different models, hitting 128 different response shapes seems possible even for not-*especially*-large APIs.

That said, I guess in almost all cases it would be fine if only the top 100 or so most commonly used response types were typically cached 😄.

It's easy enough to change later if anyone reports performance issues that could be improved by increasing it. I'll switch it to the default for now, but let me know if you think it should be set higher for any reason (I don't see a reason to set it lower than the default.)

Reply:
Actually, I think another good application of this function would be request parsing in server-side frameworks. In that case, I think you'd generally want to cache all possible parsing types for speed (given the relatively limited marginal memory requirements). So maybe it does make sense to use a higher number than  128.

Thoughts?

Reply:
My concern was someone calling this with some dynamic or semi dynamic type, e.g. an end user can directly or indirectly decide what `type_` is, and therefore create a memory leak.

Imagine a spreadsheet like application where the user can decide the types of each column, the application developer then calls `parse_as_type(Tuple[int, str, str, str, datetime], data)` (where the tuple is used to represent the types of the columns)  to validate the data. By changing the types of columns you could in theory fill up memory.

Why don't we set a maximum to 2048? That makes it unlikely the limit will ever be reached but gives a safety net.